totalRP3/modules/NamePlates/NamePlates.lua

```diff
@@ -0,0 +1,1124 @@
+	self.registerUnitIDCooldowns = {};
+	self.unitFrameWidgets = {};
+
+	self.fontStringPool = CreateFontStringPool(UIParent, "ARTWORK", 0, "SystemFont_NamePlate");
+	self.texturePool = CreateTexturePool(UIParent, "ARTWORK", 0);
+
+	-- Register configuration keys. Do this on-init so the keys always exist.
+	TRP3_Config.registerConfigKey(CONFIG_NAMEPLATES_ENABLE_CUSTOMIZATIONS, true);
+	TRP3_Config.registerConfigKey(CONFIG_NAMEPLATES_ONLY_IN_CHARACTER, true);
+	TRP3_Config.registerConfigKey(CONFIG_NAMEPLATES_ACTIVE_QUERY, true);
+	TRP3_Config.registerConfigKey(CONFIG_NAMEPLATES_SHOW_PLAYER_NAMES, true);
+	TRP3_Config.registerConfigKey(CONFIG_NAMEPLATES_SHOW_PET_NAMES, true);
+	TRP3_Config.registerConfigKey(CONFIG_NAMEPLATES_SHOW_COLORS, true);
+	TRP3_Config.registerConfigKey(CONFIG_NAMEPLATES_SHOW_ICONS, true);
+	TRP3_Config.registerConfigKey(CONFIG_NAMEPLATES_SHOW_TITLES, true);
+	TRP3_Config.registerConfigKey(CONFIG_NAMEPLATES_SHOW_OOC_INDICATORS, true);
+	TRP3_Config.registerConfigKey(CONFIG_NAMEPLATES_OOC_INDICATOR, GetDefaultOOCIndicator());
+
+	-- Prevent the module from actually being *enabled* if there's any of the
+	-- conflicting addons installed.
+	for _, addonName in ipairs(CONFLICTING_ADDONS) do
+		local state = GetAddOnEnableState(TRP3_Globals.player, addonName);
+		if state == 2 then
+			return false, format(L.NAMEPLATES_ERR_ADDON_CONFLICT, addonName);
+		end
+	end
+end
+
+-- Handler called when the module is started up. This is responsible for
+-- fully starting the module, registering events and hooks as needed.
+function TRP3_NamePlates:OnEnable()
+	-- Register events and script handlers.
+	local eventFrame = CreateFrame("Frame");
+	eventFrame:RegisterEvent("NAME_PLATE_UNIT_ADDED");
+	eventFrame:RegisterEvent("NAME_PLATE_UNIT_REMOVED");
+	eventFrame:SetScript("OnEvent", function(_, ...)
+		return self:OnEvent(...);
+	end);
+
+	-- Register internal events.
+	TRP3_Events.registerCallback("MOUSE_OVER_CHANGED", function(...)
+		self:OnMouseOverChanged(...);
+	end);
+
+	TRP3_Events.registerCallback("REGISTER_DATA_UPDATED", function(...)
+		self:OnRegisterDataUpdated(...);
+	end);
+
+	TRP3_Events.registerCallback("CONFIG_SETTING_CHANGED", function(...)
+		self:OnConfigSettingChanged(...);
+	end);
+
+	-- Install hooks.
+	hooksecurefunc("CompactUnitFrame_UpdateName", function(frame)
+		return self:OnUnitFrameNameChanged(frame);
+	end);
+
+	-- Put in a low resolution ticker to detect IC/OOC swaps. There's no event
+	-- that fires to signal this, and a similar approach is used by the
+	-- toolbar already. One second should be small enough to make it feel
+	-- like the setting works, and infrequent enough to not be an issue.
+	(function()
+		local previousState;
+		C_Timer.NewTicker(1, function()
+			local currentUser = Player.GetCurrentUser();
+			local isInCharacter = currentUser:IsInCharacter();
+
+			if previousState ~= isInCharacter then
+				previousState = isInCharacter;
+				self:OnRoleplayStatusChanged();
+			end
+		end);
+	end)();
```

Comment:
This is probably okay for now, but it probably makes sense to add the missing event.